src/radical/pilot/agent/executing/popen.py

```diff
@@ -708,33 +657,60 @@ def _get_rank_ids(self, n_ranks, launcher):
 
     # --------------------------------------------------------------------------
     #
-    def _get_post_exec(self, task):
+    def _get_exec_ranks(self, ranks, sig):
+
+        ret = 'case "$RP_RANK" in\n'
+        for rank_id, cmds in ranks.items():
+            ret += '    %d)\n' % int(rank_id)
+            for cmd in cmds:
+                # FIXME: exit on error, but don't stall other ranks on sync
+                ret += \
+                    '        %s' % cmd + \
+                    '\n' if sig == 'pre_exec' else ' %s' % self._get_check(sig)
```

Comment:
I don't find the single-line `if/else` construct not very readable in Python and would suggest to avoid it
```py
ret +=  '        %s' % cmd
if sid == 'pre_exec':
    ret += '\n'
else:
    ret += ' %s' % self._get_check(sig)
```
But it is correct, so ` ¯\_(ツ)_/¯ `